src/screens/MainScreen.tsx

```diff
@@ -13,81 +18,24 @@ import { OptionKey } from '../components/Bars/SearchBar/OptionRow';
 
 import filterCards from '../utils/filterCards';
 
-/*
-todo:
-      make search maintain proper screen composition
-*/
-
-const cards = [
-  {
-    name: 'Biedronka',
-    image: require('../assets/images/biedronka_homepage.jpg'),
-    type: 'virtual',
-  },
-  {
-    name: 'Lidl',
-    image: require('../assets/images/lidl.png'),
-    type: 'virtual',
-  },
-  {
-    name: 'Akbar brothers',
-    image: require('../assets/images/akbar.png'),
-    type: 'real',
-  },
-  {
-    name: 'Costadoro Coffee',
-    image: require('../assets/images/costadoro.png'),
-    type: 'virtual',
-  },
-  {
-    name: 'Decathlon',
-    image: require('../assets/images/decathlon.png'),
-    type: 'real',
-  },
-  {
-    name: 'Deichmann',
-    image: require('../assets/images/deichmann.png'),
-    type: 'real',
-  },
-  {
-    name: 'Espirit',
-    image: require('../assets/images/espirit.png'),
-    type: 'virtual',
-  },
-  {
-    name: 'Kaufland',
-    image: require('../assets/images/kaufland.png'),
-    type: 'real',
-  },
-  {
-    name: 'Maggi spices',
-    image: require('../assets/images/maggi.png'),
-    type: 'virtual',
-  },
-  {
-    name: 'Pollo Feliz',
+function getName(card) {
+  if (card.type === 'virtual') return card.content.businessDetails.name;
+  return card.content.name;
+}
 
-    image: require('../assets/images/pollofeliz.png'),
-    type: 'real',
-  },
-  {
-    name: 'Roblox',
-    image: require('../assets/images/roblox.png'),
-    type: 'real',
-  },
-  {
-    name: 'Stock',
-    image: require('../assets/images/stock.png'),
-    type: 'virtual',
-  },
-];
+function getImage(card) {
+  if (card.type === 'virtual') return card.content.businessDetails.bannerImageId;
+  return card.content.image;
```

Comment:
karta rzeczywista będzie miała barcode jako zdjęcie?

Reply:
Jeśli tylko w tych miejscach będą wykorzystywane te funkcje, to zrób folder `MainScreen` i plik z utilsami